pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -208,259 +180,270 @@ func (r *ReconcileRazeeDeployment) Reconcile(request reconcile.Request) (reconci
 	}
 
 	/******************************************************************************
-	CHECK THE INSTANCE FOR VALUES PASSED DOWN FROM MARKETPLACE CONFIG
-	check the instance for rhmSecretNameNonNil
-	check the instance for *clusterUUID
+	PROCEED WITH CREATING RAZEE PREREQUISITES?
 	/******************************************************************************/
-	rhmSecretName := "rhm-operator-secret"
-	clusterUUID := &instance.Spec.ClusterUUID
-	targetNamespace := *instance.Spec.TargetNamespace
-
-	if instance.Spec.DeploySecretName != nil {
-		rhmSecretName = *instance.Spec.DeploySecretName
+	if instance.Spec.DeployConfig == nil {
+		reqLogger.Info("rhm-operator-secret has not been applied")
+		return r.reconcileRhmOperatorSecret(*instance, request)
 	}
 
-	/******************************************************************************
-	CHECK FOR COMBINED SECRET
-	check for the presence of the combined secret
-	/******************************************************************************/
-	combinedSecret := corev1.Secret{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      rhmSecretName,
-		Namespace: request.Namespace,
-	}, &combinedSecret)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// Request object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			// report to status that we haven't found the secret
-			reqLogger.Info("Updating RedHatMarketplaceSecretFound")
-			instance.Status.RedHatMarketplaceSecretFound = &redHatMarketplaceSecretFound
-			*instance.Status.RedHatMarketplaceSecretFound = false
-			err = r.client.Status().Update(context.TODO(), instance)
+	// TODO: is there a way to refactor this so reconcileRhmOperatorSecret() doesn't have to be called twice.
```

Comment:
Calling it twice is fine here. The DeployConfig is created but missing values. You could move it one level above and make it like this:

```go
if instance.Spec.DeployConfig != nil && len(instance.Status.MissingDeploySecretValues) > 0 {
			reqLogger.Info("Missing required razee configuration values")
			return r.reconcileRhmOperatorSecret(*instance, request)
}
```


Place it below the block on line 188.